src/ecalc/libraries/libecalc/common/libecalc/core/graph_result.py

```diff
@@ -246,108 +266,111 @@ def get_asset_result(self) -> libecalc.dto.result.EcalcModelResult:
-                            "component_level": ComponentLevel.MODEL,
-                            "energy_usage_cumulative": model.energy_usage.to_volumes().cumulative().dict(),
-                            "power_cumulative": (
-                                model.power.to_volumes().to_unit(Unit.GIGA_WATT_HOURS).cumulative().dict()
-                                if model.power is not None
-                                else None
-                            ),
-                        },
-                    )
-                    for model in consumer_result.models
-                ]
-            )
-
-            if self.graph.get_component(consumer_id).component_type == ComponentType.COMPRESSOR:
+            if consumer_node_info.component_type in [ComponentType.COMPRESSOR, ComponentType.COMPRESSOR_SYSTEM]:
                 component = self.graph.get_component(consumer_id)
 
-                # Get temporal expression for inlet- and outlet pressures
-                inlet_pressure_eval = GraphResult.extract_pressures_from_models(
-                    energy_usage_model=component.energy_usage_model,
-                    pressure_type=CompressorPressureType.INLET_PRESSURE,
-                )
+                for model in consumer_result.models:
+                    inlet_pressure_eval = GraphResult.extract_input_compressor_pressures_from_models(
+                        energy_usage_model=component.energy_usage_model,
+                        pressure_type=CompressorPressureType.INLET_PRESSURE,
+                        name=model.name,
+                    )
+                    outlet_pressure_eval = GraphResult.extract_input_compressor_pressures_from_models(
+                        energy_usage_model=component.energy_usage_model,
+                        pressure_type=CompressorPressureType.OUTLET_PRESSURE,
+                        name=model.name,
+                    )
 
-                outlet_pressure_eval = GraphResult.extract_pressures_from_models(
-                    energy_usage_model=component.energy_usage_model,
-                    pressure_type=CompressorPressureType.OUTLET_PRESSURE,
-                )
+                    period = Period(model.timesteps[0], model.timesteps[-1])
+                    start_index, end_index = period.get_timestep_indices(self.variables_map.time_vector)
 
-                # Evaluate pressures in temporal expression
-                requested_inlet_pressure = TimeSeriesFloat(
-                    timesteps=self.timesteps,
-                    values=TemporalExpression.evaluate(inlet_pressure_eval, self.variables_map),
-                    unit=Unit.BARA,
-                )
+                    requested_inlet_pressure = TimeSeriesFloat(
+                        timesteps=self.timesteps,
+                        values=TemporalExpression.evaluate(inlet_pressure_eval, self.variables_map),
+                        unit=Unit.BARA,
+                    )[start_index : end_index + 1]
```

Comment:
@TeeeJay: Added this to get data for the correct time interval for the model (and not data for the the whole global time vector). Makes sense?

Reply:
I guess this is more or less correct, but we should be able to get the subset of the TimeSeries directly using the `TimeSeries.subset(start, end)` or `TimeSeries.subset(period)` instead of trixing and mixing with indices all over the place, if possible.

Reply:
TimeSeries(period) is added.